src/components/core/GeoChart.tsx

```diff
@@ -74,44 +109,165 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
         );
     }
 
-    private renderChart = (): React.ReactElement => {
-        const { config, chartRenderer, execution, afterRender } = this.props;
-        const chartProps = {
+    public onLegendItemClick = (item: IPushpinCategoryLegendItem) => {
+        this.setState((prevState: IGeoChartInnerState) => {
+            item.isVisible = !prevState.legendItemsEnabled[item.legendIndex].isVisible;
+            return {
+                legendItemsEnabled: set<IPushpinCategoryLegendItem[]>(
+                    [...prevState.legendItemsEnabled],
+                    item.legendIndex,
+                    { ...item },
+                ),
+            };
+        });
+    };
+
+    private getSegmentBySeries(data: Execution.IResultHeaderItem[]): IPushpinCategoryLegendItem[] {
+        const segmentItemsUniq: Execution.IResultHeaderItem[] = unionBy(data, item => {
+            return getHeaderItemName(item);
+        });
+        const segmentItemNamesUniq: string[] = segmentItemsUniq.map(
+            (item: Execution.IResultHeaderItem): string => {
+                const name: string = getHeaderItemName(item);
+                return name || EMPTY_SEGMENT_ITEM;
+            },
+        );
+        const colorPaletteMapping = getColorPaletteMapping(segmentItemNamesUniq);
```

Comment:
I don't think we need to call `getColorPaletteMapping`.
Base on [DEFAULT_COLORS](https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/visualizations/utils/color.ts#L18) and number of segment items, we can take color for it

Reply:
fixed